index.js

```diff
@@ -27,38 +27,48 @@ function fastifyView (fastify, opts, next) {
   const charset = opts.charset || 'utf-8'
   const propertyName = opts.propertyName || 'view'
   const engine = opts.engine[type]
-  const options = opts.options || {}
+  const globalOptions = opts.options || {}
   const templatesDir = opts.root || resolve(opts.templates || './')
   const lru = HLRU(opts.maxCache || 100)
   const includeViewExtension = opts.includeViewExtension || false
   const viewExt = opts.viewExt || ''
   const prod = typeof opts.production === 'boolean' ? opts.production : process.env.NODE_ENV === 'production'
   const defaultCtx = opts.defaultContext || {}
-  const layoutFileName = opts.layout
+  const globalLayoutFileName = opts.layout
 
-  if (layoutFileName && type !== 'dot' && type !== 'handlebars' && type !== 'ejs' && type !== 'eta') {
-    next(new Error('Only Dot, Handlebars, EJS, and Eta support the "layout" option'))
-    return
+  function layoutIsValid (_layoutFileName) {
+    if (type !== 'dot' && type !== 'handlebars' && type !== 'ejs' && type !== 'eta') {
+      throw new Error('Only Dot, Handlebars, EJS, and Eta support the "layout" option')
+    }
+
+    if (!hasAccessToLayoutFile(_layoutFileName, getDefaultExtension(type))) {
+      throw new Error(`unable to access template "${_layoutFileName}"`)
+    }
+    return 1
```

Comment:
why 1?

Reply:
Good catch. The whole return statement unnecessary